Dockerfile

```diff
@@ -41,36 +41,34 @@ RUN apt-get clean \
   && apt-get update \
   && apt-get install -y \
   build-essential python3.9 python3-pip git cmake make ruby rake autoconf automake \
-  && pip install conan -v "conan==1.59.0"
+  && pip install conan -v "conan==2.0.9"
 
 # make an agent directory and cd into it
 WORKDIR /root/agent
 
 # bring in the repo contents, minus .dockerignore files
 COPY . .
 
-# set some variables
-ENV PATH=$HOME/venv3.9/bin:$PATH
-ENV CONAN_PROFILE=conan/profiles/docker
-ENV WITH_RUBY=True
-ENV WITH_TESTS=False
+ARG WITH_RUBY=True
+ARG WITH_TESTS=False
 
 # limit cpus so don't run out of memory on local machine
 # symptom: get error - "c++: fatal error: Killed signal terminated program cc1plus"
 # can turn off if building in cloud
-ENV CONAN_CPU_COUNT=1
+ARG CONAN_CPU_COUNT=2
 
-# make installer
-RUN conan export conan/mqtt_cpp \
-  && conan export conan/mruby \
-  && conan install . -if build --build=missing \
-  -pr $CONAN_PROFILE \
-  -o build_tests=$WITH_TESTS \
-  -o run_tests=$WITH_TESTS \
-  -o with_ruby=$WITH_RUBY
 
-# compile source (~20mins - 4hrs for qemu)
- RUN conan build . -bf build
+# set some variables
+ENV PATH=$HOME/venv3.9/bin:$PATH
+ENV CONAN_PROFILE=conan/profiles/docker
+
+# make installer
+RUN conan profile detect
+RUN conan install . --build=missing -pr $CONAN_PROFILE -o with_ruby=$WITH_RUBY -c tools.build:skip_test=True \
+     -c tools.build:jobs=$CONAN_CPU_COUNT
+RUN conan create . --build=missing -pr $CONAN_PROFILE -o with_ruby=$WITH_RUBY -o cpack=True \
+      -o zip_destination=/root/agent -o agent_prefix=mtc \
+      -c tools.build:skip_test=True -tf "" -c tools.build:jobs=$CONAN_CPU_COUNT
```

Comment:
I think we should use  `${CONAN_CPU_COUNT:-2}` without using `ENV` in `Dockerfile`; then it could be overridden using `--build-arg CONAN_CPU_COUNT=10`.

Reply:
Currently it is an ARG that is defaulted to 2. It is not an ENV. 

```
ARG CONAN_CPU_COUNT=2
```

It can be overridden by the command line `--build-arg CONAN_CPU_COUNT=10`